src/js/controllers/oppController.js

```diff
@@ -62,149 +62,111 @@ export const oppScreen = async function(state){
+            const btn = e.target.closest('.referral');
+            if(btn) {
+                // clear right screen
+                clearScreen();
+                // render temp 2
+                oppView.renderDetail(state.opp.opps[btn.id]);
+
+                document.querySelector('.back-top').addEventListener('click', e => {
+                    const btn2 = e.target.closest('.back-top');
+
+                    if(btn) {
+                        // clear right screen
+                        clearScreen();
+                        // render temp 2
+                        billFunction(state,oppSetUp);
+                    }
+                });
+
+                //console.log("attention: " + state.user.uid);
+                //console.log("attention: " + state.opp.opps[btn.id].data().registered[1].uid);
+                let flag = false;
+                for (let i=0; i<state.opp.opps[btn.id].data().registered.length; i++){
+                    if (state.opp.opps[btn.id].data().registered[i].uid == state.user.uid){
+                        flag = true;
+                    }
                 }
-            });
-
-            //console.log("attention: " + state.user.uid);
-            //console.log("attention: " + state.opp.opps[btn.id].data().registered[1].uid);
-            let flag = false;
-            for (let i=0; i<state.opp.opps[btn.id].data().registered.length; i++){
-                if (state.opp.opps[btn.id].data().registered[i].uid == state.user.uid){
-                    flag = true;
+                if (flag){
+                    document.querySelector('.ref-request').innerHTML = `
+                    <div class="ref-request" id="1">
+                        <button type="button">Request Sent!</button>
+                    </div>
+                    `;
+                } else {
+                    document.querySelector('.ref-request').innerHTML = `
+                    <div class="ref-request" id="0">
+                        <button type="button">Request a Referral</button>
+                    </div>
+                    `;
                 }
-            }
-            if (flag){
-                document.querySelector('.ref-request').innerHTML = `
-                <div class="ref-request" id="1">
-                    <button type="button">Request Sent!</button>
-                </div>
-                `;
-            } else {
-                document.querySelector('.ref-request').innerHTML = `
-                <div class="ref-request" id="0">
-                    <button type="button">Request a Referral</button>
-                </div>
-                `;
-            }
 
 
-            document.querySelector('.ref-request').addEventListener('click', e => {
-                const btn3 = e.target.closest('.ref-request');
-                console.log("attention: "+state.opp.opps[btn.id].id);
+                document.querySelector('.ref-request').addEventListener('click', e => {
```

Comment:
You never clear your EventListeners. If I repeatedly click and unclick the button, I add more and more EventListeners.

Reply:
This is the case for all EventListeners not just here. Please do a global search for EventListeners and change as appropriate

Reply:
Problem solved.